Code/Tools/AssetProcessor/Platform/Mac/native/FileWatcher/FileWatcher_macos.cpp

```diff
@@ -93,7 +95,7 @@ void FileWatcher::WatchFolderLoop()
     static const CFTimeInterval secondsToProcess = 0.5;
 
     m_platformImpl->m_runLoop = CFRunLoopGetCurrent();
-    FSEventStreamScheduleWithRunLoop(m_platformImpl->m_stream, m_platformImpl->m_runLoop, kCFRunLoopDefaultMode);
+    FSEventStreamSetDispatchQueue(m_platformImpl->m_stream, m_platformImpl->m_dispatchQueue);
     FSEventStreamStart(m_platformImpl->m_stream);
```

Comment:
LGTM, I'm not super familiar with how this works, but in this DirectoryWatcher example, they do this work in a `DISPATCH_QUEUE_SERIAL` queue (line 220) https://clang.llvm.org/doxygen/DirectoryWatcher-mac_8cpp_source.html with this comment
```
    // We need to start watching the directory before we start scanning in order
    // to not miss any event. By dispatching this on the same serial Queue as
    // the FSEvents will be handled we manage to start watching BEFORE the
    // inital scan and handling events ONLY AFTER the scan finishes.
```